nextflow/rmatsTurbo.nf

```diff
@@ -0,0 +1,232 @@
+nextflow.enable.dsl=2
+
+
+process rmats_prep_g1 {
+  tag "rmats_prep_g1"
+  label "mid_memory"
+  publishDir path: "${params.publishDir}/read_outcomes", mode: 'copy', pattern: "prep_*_read_outcomes_by_bam.txt"
+
+  input:
+    tuple val(bam_name), file(bam)
+    each file(gtf)
+    val(group)
+
+  output:
+    path "outfd/*.rmats", emit: rmat
+    path "prep_${bam_id}_read_outcomes_by_bam.txt", emit: rob
+    val "${bam_name}", emit: bam_name
+
+  script:
+    bam_id = "${group}_${bam_name}"
+
+    read_type_value = params.is_single_end ? "single" : "paired"
+    variable_read_length_opt = params.variable_read_length ? "--variable-read-length" : ""
+    anchorLength_opt = params.anchorLength ? "--anchorLength ${params.anchorLength}" : ""
+    novelSS_opt = params.novelSS ? "--novelSS" : ""
+    mil_opt = params.novelSS ? params.mil ? "--mil ${params.mil}" : "" : ""
```

Comment:
The double `?` is a bit confusing. The WDL has `mil` and `mel` as required `Int` params. It seems reasonable to assume `mil` and `mel` are set here

Reply:
It is the same logical as WDL. I collapse the `mil_opt` and `mil_val`.